components/script/dom/htmlimageelement.rs

```diff
@@ -1528,6 +1528,15 @@ impl VirtualMethods for HTMLImageElement {
         self.super_type().unwrap().unbind_from_tree(context);
         let document = document_from_node(self);
         document.unregister_responsive_image(self);
+        if let Some(parent) = self.upcast::<Node>().GetParentElement() {
+            if parent.is::<HTMLPictureElement>() {
+                if let Some(previous_sibling) = self.upcast::<Node>().GetPreviousSibling() {
+                    if previous_sibling.is::<HTMLSourceElement> () {
```

Comment:
Is this trying to address "The element's parent is a picture element and a source element that was a previous sibling is removed."? We need to check for that state in the HTMLSourceElement::unbind_from_tree instead.

Reply:
Yes, I tried to do it here because of the discussion that we had on Gitter. Will change this 👍